### DISCUSSION TITLE
Online games

### DISCUSSION THREAD

**Discussion by francipvb** [socketio/socket.io#4384] [Q&A]
Hi all,

I want to develop a backend for an online game. I'm considering Socket.IO for the communication layer.

How it does in intensive data traffic situations like a game world?

I know that for data it is a good stack but I am not sure about game servers.

I am also considering GRPC.

**Comment by darrachequesne** [comment]
Hi! 

In most cases, Socket.IO creates a WebSocket connection between the client and the server, which is a bidirectional communication channel over TCP: https://en.wikipedia.org/wiki/WebSocket

So it will depend on your use case, maybe UDP is more suited for your game.

That being said, Socket.IO provides some useful features too: https://socket.io/docs/v4/#features

Regarding gRPC, please also check this: https://www.infoq.com/articles/websocket-and-http2-coexist/
